mlir/include/mlir/Tools/mlir-opt/MlirOptMain.h

```diff
@@ -53,25 +53,28 @@ using PassPipelineFn = llvm::function_ref<LogicalResult(PassManager &pm)>;
 /// - emitBytecode will generate bytecode output instead of text.
 /// - implicitModule will enable implicit addition of a top-level
 /// 'builtin.module' if one doesn't already exist.
-/// - dumpPassPipeline will dump the pipeline being run to stderr
-LogicalResult
-MlirOptMain(llvm::raw_ostream &outputStream,
-            std::unique_ptr<llvm::MemoryBuffer> buffer,
-            const PassPipelineCLParser &passPipeline, DialectRegistry &registry,
-            bool splitInputFile, bool verifyDiagnostics, bool verifyPasses,
-            bool allowUnregisteredDialects,
-            bool preloadDialectsInContext = false, bool emitBytecode = false,
-            bool implicitModule = false, bool dumpPassPipeline = false);
-
-/// Support a callback to setup the pass manager.
-/// - passManagerSetupFn is the callback invoked to setup the pass manager to
-///   apply on the loaded IR.
+/// - dumpPassPipeline will dump the pipeline being run to stderr.
+/// - irdlFile is the path to an IRDL file to load.
 LogicalResult MlirOptMain(
     llvm::raw_ostream &outputStream, std::unique_ptr<llvm::MemoryBuffer> buffer,
-    PassPipelineFn passManagerSetupFn, DialectRegistry &registry,
+    const PassPipelineCLParser &passPipeline, DialectRegistry &registry,
     bool splitInputFile, bool verifyDiagnostics, bool verifyPasses,
     bool allowUnregisteredDialects, bool preloadDialectsInContext = false,
-    bool emitBytecode = false, bool implicitModule = false);
+    bool emitBytecode = false, bool implicitModule = false,
+    bool dumpPassPipeline = false, StringRef irdlFile = "");
+
+/// Support a callback to setup the pass manager.
+/// - passManagerSetupFn is the callback invoked to setup the pass manager to
+///   apply on the loaded IR.
+LogicalResult MlirOptMain(llvm::raw_ostream &outputStream,
+                          std::unique_ptr<llvm::MemoryBuffer> buffer,
+                          PassPipelineFn passManagerSetupFn,
+                          DialectRegistry &registry, bool splitInputFile,
+                          bool verifyDiagnostics, bool verifyPasses,
+                          bool allowUnregisteredDialects,
+                          bool preloadDialectsInContext = false,
+                          bool emitBytecode = false,
+                          bool implicitModule = false, StringRef irdlFile = "");
```

Comment:
I am confused why the diff shows the two functions as being reordered?

Reply:
Git is being weird here, but that's because of the reformatting. Since one function got reformatted like the other one, it looks like they are being reordered, even if they don't.